src/v/cluster/archival/purger.cc

```diff
@@ -225,7 +225,10 @@ purger::collect_manifest_paths(
             continue;
         }
 
-        collected.spillover.push_back(std::move(item.key));
+        // TODO: replace with C++23 std::string_view::contains().
+        if (path.find("manifest.bin") != std::string_view::npos) {
```

Comment:
it is possible that topic name has `manifest.bin` in the name
I think that the solution is to take into account only the last component of the path

Reply:
Updated this check with better string parsing for a spillover manifest path.